kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/resolver/DependencyResolverImpl.java

```diff
@@ -57,50 +81,95 @@ public ResolutionResult deepResolve(Context<KafkaProxy> context) {
                                                                     Map<LocalRef<KafkaProxyIngress>, KafkaProxyIngress> ingresses,
-                                                                    Map<LocalRef<KafkaService>, KafkaService> clusterRefs,
-                                                                    Map<LocalRef<KafkaProtocolFilter>, KafkaProtocolFilter> filters) {
+                                                                    Map<LocalRef<KafkaService>, KafkaService> services,
+                                                                    Map<LocalRef<KafkaProtocolFilter>, KafkaProtocolFilter> filters,
+                                                                    Set<KafkaProxy> proxies) {
         VirtualKafkaClusterSpec spec = cluster.getSpec();
-        Set<LocalRef<?>> unresolvedDependencies = new HashSet<>();
-        determineUnresolvedIngresses(spec, ingresses).forEach(unresolvedDependencies::add);
-        determineUnresolvedKafkaService(spec, clusterRefs).ifPresent(unresolvedDependencies::add);
-        determineUnresolvedFilters(spec, filters).forEach(unresolvedDependencies::add);
-        return new ClusterResolutionResult(cluster, unresolvedDependencies);
+        LocalRef<VirtualKafkaCluster> clusterRef = toLocalRef(cluster);
+        Set<UnresolvedReference> unresolved = new HashSet<>();
+        determineUnresolvedIngresses(clusterRef, spec, ingresses).forEach(unresolved::add);
+        determineIngressesWithUnresolvedProxies(ingresses, proxies).forEach(unresolved::add);
+        determineUnresolvedKafkaProxies(clusterRef, spec, proxies).ifPresent(unresolved::add);
+        determineUnresolvedKafkaService(clusterRef, spec, services).ifPresent(unresolved::add);
+        determineUnresolvedFilters(clusterRef, spec, filters).forEach(unresolved::add);
+        Set<KafkaProxyIngress> ingressesWithResolvedRefsFalse = ingresses.values().stream()
+                .filter(i -> hasAnyResolvedRefsFalse(Optional.ofNullable(i.getStatus()).map(KafkaProxyIngressStatus::getConditions).orElse(List.of())))
+                .collect(Collectors.toSet());
+        Set<KafkaProtocolFilter> filtersWithResolvedRefsFalse = filters.values().stream()
+                .filter(i -> hasAnyResolvedRefsFalse(Optional.ofNullable(i.getStatus()).map(KafkaProtocolFilterStatus::getConditions).orElse(List.of())))
+                .collect(Collectors.toSet());
+        Set<KafkaService> kafkaServicesWithResolvedRefsFalse = services.values().stream()
+                .filter(i -> hasAnyResolvedRefsFalse(Optional.ofNullable(i.getStatus()).map(KafkaServiceStatus::getConditions).orElse(List.of())))
+                .collect(Collectors.toSet());
+        return new ClusterResolutionResult(cluster,
+                new ResolutionResult.UnresolvedReferences(unresolved, ingressesWithResolvedRefsFalse, filtersWithResolvedRefsFalse, kafkaServicesWithResolvedRefsFalse));
```

Comment:
need to unit test this new Resolver functionality around detecting resolved refs false